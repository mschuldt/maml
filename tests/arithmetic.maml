arduino = Arduino(desktop=True)

@block(once)
def test():
    # True assignments
    print(1+1 == 2) # 1
    print(1+1 != 0) # 1
    print(1+1 <= 2) # 1
    print(1+1 <= 3) # 1
    print(1+1 < 3)  # 1
    print(1+1 >= 2) # 1
    print(1+1 > 1)  # 1
    # False assignments
    print(1+1 != 2) # 0
    print(1+1 == 0) # 0
    print(1+1 >= 3) # 0
    print(1+1 > 3)  # 0
    print(1+1 <= 1) # 0
    print(1+1 < 1)  # 0
    print("s" == "s")
    # Arithmetic equality tests
    print(1 - 1) # 0
    print(2 - 1) # 1
    print(1 - 2) # -1
    print(4 * 6) # 24
    print(6 * 4) # 24
    print(4 * 0) # 0
    print(0 * 4) # 0
    print(6 / 3) # 2
    print(6 / 4) # 1
    print(0 / 4) # 0

    print(-1 + 2) # 1
    die()

arduino.send(test)
