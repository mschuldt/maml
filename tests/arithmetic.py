arduino = Arduino(desktop=True)

@block
def test():
    # True assignments
    print_i(1+1 == 2) # 1
    print_i(1+1 != 0) # 1
    print_i(1+1 <= 2) # 1
    print_i(1+1 <= 3) # 1
    print_i(1+1 < 3)  # 1
    print_i(1+1 >= 2) # 1
    print_i(1+1 > 1)  # 1
    # False assignments
    print_i(1+1 != 2) # 0
    print_i(1+1 == 0) # 0
    print_i(1+1 >= 3) # 0
    print_i(1+1 > 3)  # 0
    print_i(1+1 <= 1) # 0
    print_i(1+1 < 1)  # 0
    print_i("s" == "s")
    # Arithmetic equality tests
    print_i(1 - 1) # 0
    print_i(2 - 1) # 1
    print_i(1 - 2) # -1
    print_i(4 * 6) # 24
    print_i(6 * 4) # 24
    print_i(4 * 0) # 0
    print_i(0 * 4) # 0
    print_i(6 / 3) # 2
    print_i(6 / 4) # 1
    print_i(0 / 4) # 0

    print_i(-1 + 2) # 1
    die()

arduino.send(test)
