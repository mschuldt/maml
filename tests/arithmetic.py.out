Initializing avm...
Ready.

1
1
1
1
1
1
1
0
0
0
0
0
0
0
1
-1
24
24
0
0
2
1
0
1
bye.
